test/integration_tests.jl

```diff
@@ -31,3 +31,21 @@ end
     @test record.initial_sizes == [(32, 64), (32, 2)]
     @test (@ballocated avoid_allocations($record, f_matmul_noprealloc)) <= 1520
 end
+
+@testset "check thread-safe" begin
```

Comment:
Need to also change the .travis.ci to make it actually use multiple threads, via setting the enviroment variable